lodash_exercise/lodash.js

```diff
@@ -1,85 +1,263 @@
+}
 
+// =============================================================================
+function takeRight(array, n = 1){
+  let begin = array.length - n < 0 ? 0 : array.length - n;
+  return array.slice(begin);
 }
 
+// =============================================================================
 function union(){
+  let arrays = Array.from(arguments);
+  let result = [];
+
+  for (let i = 0; i < arrays.length; i++) {
+    for (let j = 0; j < arrays[i].length; j++) {
+      console.log(arrays[i][j]);
+      if(!result.includes(arrays[i][j])) result.push(arrays[i][j]);
+    }
+  }
 
+  return result;
 }
 
-function zipObject(){
+// =============================================================================
+function zipObject(keys, values){
+  let result = {};
+
+  for (let i = 0; i < keys.length; i++) {
+    result[keys[i]] = values[i];
+  }
 
+  return result;
 }
 
-function includes(){
+// =============================================================================
+function includes(collection, value, fromIndex = 0){
+  if (Array.isArray(collection)) {
+    if (collection.indexOf(value) === fromIndex) return true;
+    else return false;
 
-}
+  } else if (typeof collection === 'object') {
+    for (let key in collection) {
+      if (collection[key] === value) return true;
+    }
 
-function sample(){
+    return false;
+  } else if (typeof collection === 'string') {
+    if (collection.indexOf(value) !== -1) return true;
+    else return false;
+  }
 
+  return false;
 }
 
-function cloneDeep(){
+// =============================================================================
+function sample(collection){
+  let rnd = Math.floor(Math.random() * collection.length);
+
+  if (Array.isArray(collection)) return collection[rnd];
+  else {
+    let keyNum = 0;
 
+    for (let key in collection) {
+      if (keyNum === rnd) return collection[key];
+      keyNum++;
+    }
+  }
 }
 
-function sumBy(){
+// =============================================================================
+function cloneDeep(value){
+  let clone = [];
+
+  function helper(val) {
+    if (Array.isArray(val)) {
+      for (var i = 0; i < val.length; i++) {
+        clone.concat(cloneDeep(val[i]));
+      }
 
+      return clone;
+    }
+  }
+
+  helper(value);
+
+  return clone;
 }
 
-function inRange(){
+// =============================================================================
+function sumBy(array, iteratee){
+  let sum = 0;
+
+  if (typeof iteratee === 'string') {
+    for (let i = 0; i < array.length; i++) {
+      if (array[i][iteratee]) sum += array[i][iteratee];
+    }
+  }
 
+  if (typeof iteratee === 'function') {
+    for (let i = 0; i < array.length; i++) {
+      sum += iteratee(array[i]);
+    }
+  }
+
+  return sum;
 }
 
-function has(){
+// =============================================================================
+function inRange(number, end, start = 0){
+  if (start > end) {
+    let swap = start;
+    start = end;
+    end = swap;
```

Comment:
if you're using ES2015 default parameters, try to use ES2015 array destructuring to swap values `[start, end] = [end,start]` - we'll cover that more on friday